pykeepass/pykeepass.py

```diff
@@ -152,14 +176,19 @@ def save(self, filename=None, transformed_key=None):
         if not filename:
             filename = self.filename
 
+        if hasattr(self.kdbx.header, 'data'):
```

Comment:
I noticed that if you called `save()` after changing anything in the header it didn't really save because the binary representation was already there...